qutebrowser/components/adblock.py

```diff
@@ -147,7 +147,20 @@ def _read_hosts_file(self, filename: str, target: typing.Set[str]) -> bool:
         try:
             with open(filename, 'r', encoding='utf-8') as f:
                 for line in f:
-                    target.add(line.strip())
+                    parts = line.strip().split()
+                    if len(parts) == 1:
+                        # "one host per line" format
+                        hosts = parts[:1]
+                    else:
+                        # /etc/hosts format
+                        hosts = parts[1:]
+                    
+                    for host in hosts:
+                        if ('.' in host and
```

Comment:
This check is duplicated in two places now, could you lift it into a helper function instead?

Reply:
Pretty much the entire logic is duplicated from `_parse_line`, not just the check. I don't have time for a deeper look at the moment, but it should be possible to re-use that.